SPECS/tdnf/virtual-repo-snapshot.patch

```diff
@@ -0,0 +1,906 @@
++        pTracking->nPrintPackage = 0;
++        pTracking->nTimeFound = 0;
++    }
++cleanup:
++    if (pszElementBuffer)
++    {
++        free(pszElementBuffer);
++    }
++    return;
++error:
++    goto cleanup;
++}
++
++void TDNFFilterCharDataHandler(void * userData, const char * content, int length) {
++    uint32_t dwError = 0;
++    // load tracking data
++    TDNFFilterData * pTracking = (TDNFFilterData *)userData;
++    pTracking->nPrevElement = 1;
++
++    char * pszCharData = malloc((length + 1) * sizeof(char));
++    if (!pszCharData)
++    {
++        dwError = ERROR_TDNF_TIME_FILTER_MEMORY;
++        BAIL_ON_TDNF_TIME_FILTER_ERROR(dwError);
++    }
++    bzero(pszCharData, (length + 1) * sizeof(char));
++    strncpy(pszCharData, content, length);
++    char * pszLintedCharData = descEscapeCharLinter(pszCharData);
++    if (!pszLintedCharData)
++    {
++        dwError = ERROR_TDNF_TIME_FILTER_MEMORY;
++        BAIL_ON_TDNF_TIME_FILTER_ERROR(dwError);
++    }
++
++    // check params 
++    if (!pTracking->nInPackage || pTracking->nPrintPackage)
++    {
++        // print to file
++        fprintf(pTracking->pbOutfile, "%s", pszLintedCharData);
++    }
++    else if (pTracking->nInPackage && !pTracking->nTimeFound)
++    {
++        // add to buffer
++        checkAndResizeBuffer(&(pTracking->pszElementBuffer), &(pTracking->nBufferMaxLen), length);
++        strcat(pTracking->pszElementBuffer, pszLintedCharData);
++    } // else do nothing (skipped package)
++
++cleanup:
++    if (pszLintedCharData)
++    {
++        free(pszLintedCharData);
++    }
++    if (pszCharData)
++    {
++        free(pszCharData);
++    }
++    return;
++error:
++    goto cleanup;
++}
++
++char * 
++TDNFFilterFile(const char * pszInFilePath, char * pszEpochTime) {
++    // vars
++    uint32_t dwError = 0;
++    TDNFFilterData * pData = NULL;
++    FILE * pbInFile = NULL;
++    FILE * pbOutFile = NULL;
++    XML_Parser bParser;
++    bzero(&bParser, sizeof(XML_Parser));
++    char pszTimeExtension[100];
++    char * pszOutFilePath = NULL;
++    time_t nEpochTime;
++    bzero(&nEpochTime, sizeof(time_t));
++    char * pszEpochEnd = NULL;
++    nEpochTime = strtoll(pszEpochTime, &pszEpochEnd, 10);
++    
++    if (nEpochTime == 0 && strcmp(pszEpochTime, "0") != 0) {
++        printf("Error converting --epochtimelimit: %s\n", pszEpochTime);
++        dwError = ERROR_TDNF_TIME_FILTER_PARSE;
++        BAIL_ON_TDNF_TIME_FILTER_ERROR(dwError);
++    }
++    if (pszEpochEnd != NULL && strcmp(pszEpochEnd, "") != 0) {
++        printf("The time provided to --epochtimelimit is misformatted: \'%s\' should be an epoch time\n", pszEpochTime);
++        dwError = ERROR_TDNF_TIME_FILTER_PARSE;
++        BAIL_ON_TDNF_TIME_FILTER_ERROR(dwError);
++    }
++
++    sprintf(pszTimeExtension, "-%ld.xml", nEpochTime);
++    int nOutFileLen = strlen(pszInFilePath) + strlen(pszTimeExtension);
++
++    pszOutFilePath = malloc(nOutFileLen * sizeof(char));
```

Comment:
Didn't leave space for the null-character

Reply:
Since the InFilePath does not have the file extension + compression cleaved off it will technically have enough space, but we can do better/be more exact. I have moved this calculation down and added 1 for the null char specifically.